src/platform_linux/platform_types.h

```diff
@@ -105,7 +105,10 @@ platform_assert_msg(platform_stream_handle stream,
 typedef pthread_t platform_thread;
 
 // Mutex
-typedef pthread_mutex_t platform_mutex;
+typedef struct {
```

Comment:
Maybe [?] expand on the lonely `Mutex` comment to something like:

`Thread-specific mutex, with ownership tracking.` ?